src/ctia/entity/event.clj

```diff
@@ -62,22 +69,89 @@
 
 (def EventGetParams EventFieldsParam)
 
+(defn same-bucket? [bucket event]
+  (let [max-seconds (get-in @properties [:ctia :events :timeline :max-seconds] 5)
+        from        (t/minus (:from bucket) (t/seconds max-seconds))
+        to          (t/plus (:to bucket) (t/seconds max-seconds))]
+    (and (= (:owner bucket) (:owner event))
+         (t/before? from (:timestamp event))
+         (t/before? (:timestamp event) to))))
+
+(defn init-bucket [event]
+  {:count 1
+   :owner (:owner event)
+   :from (:timestamp event)
+   :to (:timestamp event)
+   :events (list event)})
```

Comment:
I would have appreciated a schema for a `Bucket` and for a `Timeline`.
And a bit more schema annotations in function declarations.